src/ConstructiveSolidGeometry/VolumePrimitives/Cone.jl

```diff
@@ -73,13 +73,54 @@ from 2 at the bottom to 4 at the top circular base.
     
 See also [Constructive Solid Geometry (CSG)](@ref).
 """
-@with_kw struct Cone{T,CO,TR,TP<:Union{Nothing,T}} <: AbstractVolumePrimitive{T, CO}
-    r::TR = 1
-    φ::TP = nothing
-    hZ::T = 1
+struct Cone{T,CO,TR,TP<:Union{Nothing,T}} <: AbstractVolumePrimitive{T, CO}
+    r::TR 
+    φ::TP
+    hZ::T 
 
-    origin::CartesianPoint{T} = zero(CartesianPoint{T})
-    rotation::SMatrix{3,3,T,9} = one(SMatrix{3, 3, T, 9})
+    origin::CartesianPoint{T} 
+    rotation::SMatrix{3,3,T,9}
+end
+
+#Type conversion happens here
+function Cone{T,CO}(r, φ, hZ, origin, rotation) where {T,CO}
+    _r = _csg_convert_args(T, r)
+    _φ = _csg_convert_args(T, φ)
+    _hZ = _csg_convert_args(T, hZ)
+    Cone{T,CO,typeof(_r),typeof(_φ)}(_r, _φ, _hZ, origin, rotation)
+end
+
+#Type promotion happens here
+function Cone(CO, r::TR, φ::TP, hZ::TZ, origin::PT, rotation::ROT) where {TR, TP<:Real, TZ, PT, ROT}
+    eltypes = _csg_get_promoted_eltype.((TR, TZ, TP, PT, ROT))
+    T = promote_type(eltypes...)
+    Cone{T,CO}(r, φ, hZ, origin, rotation)
+end
+
+function Cone(CO, r::TR, φ::Nothing, hZ::TZ, origin::PT, rotation::ROT) where {TR, TZ, PT, ROT}
+    eltypes = _csg_get_promoted_eltype.((TR, TZ, PT, ROT))
+    T = promote_type(eltypes...)
+    Cone{T,CO}(r, φ, hZ, origin, rotation)
+end
```

Comment:
... which then also requires that we change lines 96 and 102 to 
```julia
T = float(promote_type(eltypes...))
```
such that `CSG.Cone()` yields a `Cone` of type `Float` (`Float64` to be more specific).

Reply:
This should be added anyhow, because if everything is passed as `Int`, the current version results in an error:
```julia
CSG.Cone(r = 1, hZ = 1, origin = CartesianVector(1,0,0), rotation = [1 0 0; 0 1 0; 0 0 1])
```
```
ERROR: TypeError: in AbstractGeometry, in T, expected T<:AbstractFloat, got Type{Int64}
Stacktrace:
 [1] SolidStateDetectors.ConstructiveSolidGeometry.Cone(CO::Type, r::Int64, φ::Nothing, hZ::Int64, origin::CartesianVector{Int64}, rotation::Matrix{Int64})
   @ SolidStateDetectors.ConstructiveSolidGeometry ~/SolidStateDetectors.jl/src/ConstructiveSolidGeometry/VolumePrimitives/Cone.jl:103
 [2] #Cone#339
   @ ~/SolidStateDetectors.jl/src/ConstructiveSolidGeometry/VolumePrimitives/Cone.jl:113 [inlined]
 [3] top-level scope
   @ REPL[3]:1
```
because this would try to build a `Cone{Int64}`, which (by definition) is not allowed.